### PR TITLE
registry: add cert DNS challenges, update domain point of contact

### PR DESCRIPTION
**Pull request by BenTheElder**
```
$ gcloud certificate-manager dns-authorizations --project=k8s-infra-oci-proxy list
NAME                             DOMAIN                   DNS_RECORD                                RECORD_TYPE  DNS_VALUE
registry-sandbox-k8s-io-dnsauth  registry-sandbox.k8s.io  _acme-challenge.registry-sandbox.k8s.io.  CNAME        893400bf-1899-4ca4-8cc3-1599310bb39b.12.authorize.certificatemanager.goog.

$ gcloud certificate-manager dns-authorizations --project=k8s-infra-oci-proxy-prod list
NAME                     DOMAIN           DNS_RECORD                        RECORD_TYPE  DNS_VALUE
registry-k8s-io-dnsauth  registry.k8s.io  _acme-challenge.registry.k8s.io.  CNAME        bc87a114-671a-4ff9-8b32-c93230aa3136.14.authorize.certificatemanager.goog.
```

### CODE REVIEW

File: dns/zone-configs/k8s.io._0_base.yaml [hunk 1]
```diff
@@ -262,19 +262,27 @@ monitoring.prow-canary:
 prs:
   type: CNAME
   value: redirect.k8s.io.
-# Sandbox OCI redirector service.  (@ameukam,@BenTheElder,@thockin, @eddiezane)
+# Sandbox OCI redirector service. (@ameukam,@BenTheElder,@thockin,@dims)
```
Comment (BenTheElder) [inline dns/zone-configs/k8s.io._0_base.yaml:265]: cc @dims

File: dns/zone-configs/k8s.io._0_base.yaml [hunk 2]
```diff
@@ -262,19 +262,27 @@ monitoring.prow-canary:
...
 registry-sandbox:
   - type: A
     value: 34.149.65.206
   - type: AAAA
     value: "2600:1901:0:4e2f::"
-# Prod OCI redirector service.  (@ameukam,@BenTheElder,@thockin, @eddiezane)
+# DNS challenge for redirector sandbox certs
+_acme-challenge.registry-sandbox:
+  - type: CNAME
+    value: 893400bf-1899-4ca4-8cc3-1599310bb39b.12.authorize.certificatemanager.goog.
```
Comment (BenTheElder) [inline dns/zone-configs/k8s.io._0_base.yaml:274]:
```
$ gcloud certificate-manager dns-authorizations --project=k8s-infra-oci-proxy list
NAME                             DOMAIN                   DNS_RECORD                                RECORD_TYPE  DNS_VALUE
registry-sandbox-k8s-io-dnsauth  registry-sandbox.k8s.io  _acme-challenge.registry-sandbox.k8s.io.  CNAME        893400bf-1899-4ca4-8cc3-1599310bb39b.12.authorize.certificatemanager.goog.
```

File: dns/zone-configs/k8s.io._0_base.yaml [hunk 3]
```diff
@@ -262,19 +262,27 @@ monitoring.prow-canary:
...
 # Reach to k8s-infra-oci-proxy-admins@kubernetes.io for major issues.
 registry:
   - type: A
     value: 34.107.244.51
   - type: AAAA
     value: "2600:1901:0:1013::"
+# DNS challenge for prod redirector certs
+_acme-challenge.registry:
+  - type: CNAME
+    value: bc87a114-671a-4ff9-8b32-c93230aa3136.14.authorize.certificatemanager.goog.
```
Comment (BenTheElder) [inline dns/zone-configs/k8s.io._0_base.yaml:285]:
```
$ gcloud certificate-manager dns-authorizations --project=k8s-infra-oci-proxy-prod list
NAME                     DOMAIN           DNS_RECORD                        RECORD_TYPE  DNS_VALUE
registry-k8s-io-dnsauth  registry.k8s.io  _acme-challenge.registry.k8s.io.  CNAME        bc87a114-671a-4ff9-8b32-c93230aa3136.14.authorize.certificatemanager.goog.
```

File: dns/zone-configs/k8s.io._0_base.yaml (hunk 1 above)
Comment (dims) [reply dns/zone-configs/k8s.io._0_base.yaml:265]: thanks!

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-dns/1655721973612810240 🟢

**Pull request merged**